src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs

```diff
@@ -128,6 +129,7 @@ public override void RemoveProject(ProjectId projectId)
             public override bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs e)
             {
                 return e.Option.Feature == nameof(SimplificationOptions) ||
+                       e.Option.Feature == nameof(CodeStyleOptions) ||
```

Comment:
currently third party analyzer can't use this ability. they will be able once editorconfig move down to compiler layer. otherwise, only option they have is through ruleset file.